source/grid/grid_in.cc

```diff
@@ -757,6 +757,31 @@ void GridIn<dim, spacedim>::read_ucd (std::istream &in)
   tria->create_triangulation_compatibility (vertices, cells, subcelldata);
 }
 
+namespace
+{
+  // Forward declaration of helper class
+  template <int dim>
+  class Abaqus_to_UCD;
```

Comment:
I don't see a particular problem with having the whole class here. It logically fits with the function that makes use of it right afterwards. To me, the organization of .cc files has become a bit unimportant since I switched to an IDE that can jump from one function to the other without me having to know where it is in the file :-)

But I don't feel strongly one way or the other. If you like it better this way, just keep it this way.